### PR TITLE
Remove unused macros from the codebase

### PR DESCRIPTION
**Pull request by est31**
Thanks to the lint I've implemented in #41907 I've found some unused macros inside the rustc codebase.

### CODE REVIEW

**Comment by est31** [comment]
I want to get this through separate review as its easy and simple, and to not burden @jseyfried . This change does block the other PR though, as if the lint gets added, we emit warnings for each of these macros and all of the crates have `#![deny(warnings)]`.

File: src/librustc_trans/macros.rs [hunk 1]
```diff
@@ -1,29 +0,0 @@
...
-// file at the top-level directory of this distribution and at
-// http://rust-lang.org/COPYRIGHT.
-//
-// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
-// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
-// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
-// option. This file may not be copied, modified, or distributed
-// except according to those terms.
-
-macro_rules! unpack_datum {
```
Comment (eddyb) [inline src/librustc_trans/macros.rs]: Ahahahahahhaahahaha I didn't even realize this file *existed*.

**Pull request merged**